src/executor.jl

```diff
@@ -5,19 +5,25 @@ An executor is a realization of a symbolic architecture defined by a `SymbolicNo
 The actual forward and backward computation specified by the network architecture can
 be carried out with an executor.
 """
-mutable struct Executor
+const VecOfNDArray = AbstractVector{<:NDArray}
+mutable struct Executor{A<:VecOfNDArray,
+                        B<:VecOfNDArray,
+                        G<:AbstractVector{<:Union{Void,NDArray}},
+                        O<:VecOfNDArray,
+                        D<:Dict{Symbol,<:NDArray},
+                        E<:Dict{Symbol,<:NDArray}}
```

Comment:
This is terrible, any better way to deal with abstract container type?
Or we need to redesign `Executor`?